CRM/Utils/PDF/Utils.php

```diff
@@ -72,7 +88,7 @@ public static function html2pdf($text, $fileName = 'civicrm.pdf', $output = FALS
 <html>
   <head>
     <meta http-equiv=\"Content-Type\" content=\"text/html; charset=utf-8\"/>
-    <style>@page { margin: {$t}{$metric} {$r}{$metric} {$b}{$metric} {$l}{$metric}; }</style>
+    <style>@page { margin: {$t}{$metric} {$r}{$metric} {$b}{$metric} {$l}{$metric}; $css_page_size }</style>
```

Comment:
The test fail in CRM_Contribute_Form_Task_PDFLetterCommonTest::testAllContributionTokens is because it's not expecting the extra blank space here. Maybe just include a space at the beginning in line 69 above and then no space here?

The other test I'm not sure, might just be a timeout weirdness?

Speaking of line 69, I think css itself supports the possible values that $paperSize['metric'] might be (in, cm, mm, pt, px, etc) but I guess the question is whether weasyprint does. If it doesn't then one alternate is use the $paper_height, $paper_width that's already been converted to pt and always use pt.

Reply:
Practically I went with `a4 {$orientation}` until today.  However, with all those special Avery formats for the labels... I didn't want to make a whitelist and fall back to specific units. That'd be yet another can of worms.

Regarding support in theory:
Length Units: https://doc.courtbouillon.org/weasyprint/stable/api_reference.html#css-values-and-units-module-level-3
> Font-related lengths (em, ex, ch, rem), absolute lengths (cm, mm, q, in, pt, pc, px), angles (rad, grad, turn, deg), resolutions (dpi, dpcm, dppx) are supported.

Page sizes: https://doc.courtbouillon.org/weasyprint/stable/api_reference.html#css → https://drafts.csswg.org/css-page-3/#page-size-prop